draft-beck-tls-trust-anchor-ids.md

```diff
@@ -111,30 +111,29 @@ Additionally, to support TLS clients with many trusted certification authorities
 
 # Introduction
 
-TLS {{!RFC8446}} endpoints typically authenticate using X.509 certificates {{!RFC5280}}. These are assertions by a certification authority (CA) that associate a TLS key with DNS names or other identifiers. If the peer trusts the CA, it will accept this association. The authenticating party (usually the server) is known as the *subscriber* and the peer (usually the client) is the *relying party*.
+TLS {{!RFC8446}} authentication uses X.509 certificates {{!RFC5280}} to associate the authenticating party's, or *subscriber's*, TLS key with its application identifiers, such as DNS names. These associations are signed by some certificate authority (CA). The peer, or *relying party*, curates a set of CAs that are trusted to only sign correct associations, which allows it to rely on the TLS to authenticate application identifiers. Typically the subscriber is the server and the relying party is the client.
 
-{{Section 4.2.4 of RFC8446}} defines the `certificate_authorities` extension, which allows the subscriber, who may have multiple certificates available, to select the correct certificate for the relying party. However, this extension’s size is impractical for some applications.
+A single subscriber may need to interoperate with relying parties that trust different sets of CAs. To accommodate this, {{Section 4.2.4 of RFC8446}} defines the `certificate_authorities` extension. This allows the subscriber to provision multiple certificates and select the one that will allow the relying party to accept its TLS key. This is analogous to parameter negotiation elsewhere in TLS.
 
-Without such a negotiation mechanism, the subscriber must somehow obtain certificates which simultaneously satisfy all relying parties. This is a challenge for subscribers when relying parties are diverse. This translates to analogous challenges for CAs and relying parties:
+However, this extension's size is impractical for some applications. Existing PKIs may have many CAs, and existing CAs may have long X.509 names. As of August 2023, the Mozilla CA Certificate Program {{MOZILLA-ROOTS}} contained 144 CAs, with an average name length of around 100 bytes. While applications could mitigate this with smaller PKIs and transitioning to newer CAs with shorter names, this is incompatible with existing PKIs.
```

Comment:
I'm not sure we want to address it here, but there is another challenge with the existing certificate authorities extension. 

Currently, with path building and cross signs potentially in the mix with the CA extension if a client asks for CA "Fred" and a server is in possession of a cross signed gordian knot that could be signed by "Fred" "George" "Eugene" and "Festus", The server now must implement path building, find out if their bundle satisfies CA "Fred", and if it does then either make the choice to serve up the path built result, or send the original bundle and trust in prayer and clean living that the client will path build and find "Fred" rather than something else due to inconsistencies and errors in path building implementations. 

Do we want to mention this here or sweep it under the rug for the moment?

Reply:
(also fine addressing this separately from this re-word)

Reply:
Also s/smaller PKIs" "Trusting fewer CAs" - to match the follow on text about transitioning to newer CA's.

Reply:
I don't think path-building on the server is related to the extension and should be mentioned. That's purely how you decided to represent your server configuration. Yes, OpenSSL servers sometimes path-build to compute their own configuration, and we're envisioning that, with the ACME mechanism, servers would not do this. But this has nothing to do with the negotiation scheme:

* A server that uses `certificate_authorities` could well be configured with pre-built paths and only check the last certificate against the extension.

* Conversely, a server that uses `trust_anchors` could well take a bag of certificates and path-build on the fly to expand that out into a bunch of candidate paths to send.

Path-building discussions are mostly a distraction here.